cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTable.java

```diff
@@ -590,6 +602,24 @@ private void upsertInternal(Collection<Field<?>> fields) throws IOException {
     }
   }
 
+  private void updateRangeInternal(Range keyRange, Collection<Field<?>> fields) throws IOException {
+    String sqlQuery = getUpdateRangeSqlQuery(keyRange, fields);
+    try (PreparedStatement statement = connection.prepareStatement(sqlQuery)) {
+      int index = 1;
+      // set field values to update
+      for (Field<?> field : fields) {
+        setField(statement, field, index);
+        index++;
+      }
+      setStatementFieldByRange(keyRange, statement, index);
+      LOG.trace("SQL statement: {}", statement);
+      statement.executeUpdate();
+    } catch (SQLException e) {
+      throw new IOException(String.format("Failed to write to table %s with fields %s",
```

Comment:
nit: could make the error message more precise: `String.format("Failed to update table %s with range %s and fields %s",tableSchema.getTableId().getName(), keyRange, fields)`